### PR TITLE
Add school frontend CRUD, includes create school, edit school, and delete school

### LINKED ISSUE #1: EPIC: Frontend for Schools CRUD (some of which can be done in parallel)
# User Story

* As an admin
* I can do Create/Read/Update/Destroy operations on schools
* So that I can manage the Schools

### Discussion

The backend CRUD for the schools table is already implemented;  you can find the code in the controllers, and test them on either localhost or swagger.  However, as of right now, the frontend is not there.  

Your job is to implement a frontend as in team03 for the schools table.

### Acceptance Criteria

- [ ] There is a menu option for an admin to get to the schools index page where all of the schools are listed
- [ ] There is an option on that page to create a new school
- [ ] There is an edit button for schools that takes me to school edit page where I can edit schools
- [ ] There is an delete button for schools that deletes a school

### Implementation Todos

**F1: Frontend School Implementation**

F1.1: Building create button
- [ ] Make the "create school" button work.
- [ ] Link the create school button to correct API and make a SchoolCreate.js page for that.
- [ ] Make sure it's actually creating a school

F1.2: Building Edit button
- [ ] Link the edit school button to correct API and make a SchoolEdit.js page for that.
- [ ] Make sure it actually allows you to update a school

F1.3: Building Delete button
- [ ] Delete button appears
- [ ] Link the Delete school button to correct API.
- [ ] Make sure it's actually allows you to delete a school

**B1: Bugs in Backend:**

B1.1: Update API
- [ ] You should not be able to update to the school abbrev if it used as the primary key.  If it is not used as the primary key perhaps it should be (similar to UCSBOrganizations in team03).

### PR DESCRIPTION
**Pull request by JUstISAGE**
In this pull request, I implemented EPIC: Frontend for Schools CRUD, which includes the create of school page, edit of school page and delete of school page 
dokku: https://proj-school-daniel-dev.dokku-09.cs.ucsb.edu/
Closes #1

### CODE REVIEW

**Review by Peak1260 — DISMISSED** [review]
LGTM

**Comment by pconrad** [comment]
It's not working for me on https://proj-school-daniel-dev.dokku-09.cs.ucsb.edu/

This is what I'm seeing:

create-school

File: src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java [hunk 1]
```diff
@@ -96,7 +95,7 @@ public School getById(
 
     @Operation(summary= "Delete a school")
     @PreAuthorize("hasRole('ROLE_ADMIN')")
-    @DeleteMapping("")
+    @DeleteMapping("/delete")
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java]:
```suggestion
    @DeleteMapping("")
```

This is fine. There is not need to add `/delete`

File: src/test/java/edu/ucsb/cs156/organic/controllers/SchoolControllerTests.java [hunk 2]
```diff
@@ -136,7 +136,7 @@ public void test_that_logged_in_user_can_get_by_id_when_the_id_exists() throws E
     public void test_that_logged_in_user_can_get_by_id_when_the_id_does_not_exist() throws Exception {
         when(schoolRepository.findById(eq("umn"))).thenReturn(Optional.empty());
 
-        MvcResult response = mockMvc.perform(get("/api/schools").param("abbrev", "umn")).andExpect(status().isNotFound())
+        MvcResult response = mockMvc.perform(get("/api/schools/get").param("abbrev", "umn")).andExpect(status().isNotFound())
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/organic/controllers/SchoolControllerTests.java]: Please restore the old code.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove:
* the commented out code
* the unnecessary changes to the backend.  They are not needed, and even if they were, they should be in a separate PR (but they are NOT needed.)

Finally: it isn't working for me on dokku.  So even if the issues above are addressed, you also need to figure out why it isn't working.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please fix the title also.

see: https://ucsb-cs156.github.io/topics/pull_requests/#pr-titles

**Comment by rtzhang1x** [comment]
> It's not working for me on https://proj-school-daniel-dev.dokku-09.cs.ucsb.edu/
> 
> This is what I'm seeing:
> 
> create-school

image

The same things happens in my localhost. Above is the error that occured. Maybe the brackets or the backslash (which is usually used as an escape character) is causing the illegal argument exception.

**Comment by rtzhang1x** [comment]
> > It's not working for me on https://proj-school-daniel-dev.dokku-09.cs.ucsb.edu/
> > This is what I'm seeing:
> > create-school
> 
> image
> 
> The same things happens in my localhost. Above is the error that occured. Maybe the brackets or the backslash (which is usually used as an escape character) is causing the illegal argument exception.

Caused by these lines?
image

Looking at school.java, it turns out `school.getTermRegex()` **DOES NOT EXIST**!!!

**Review by rtzhang1x — CHANGES_REQUESTED** [review]
Right now is to figure out how to bypass the invalid character error and create an actual implementation for `school.getTermRegex()`. These are the breaking parts of this pull request right now.
After that, we don't even know if your table interface is functional right now.

**Comment by JUstISAGE** [comment]
> Right now is to figure out how to bypass the invalid character error and create an actual implementation for `school.getTermRegex()`. These are the breaking parts of this pull request right now. After that, we don't even know if your table interface is functional right now.

I encodes the url manually and solved this problem, but a frontend test failed for some reasons

**Comment by Crossover661** [comment]
Looks good to me.

**Review by crosscountrycoder — COMMENTED** [review]
Looks good to me.

**Pull request merged**